nigelObj improvements

Nabarb opened the pull request:
Incremental update and small bugfixes. Can you please triple check checkAnimalsForClones? I've done enough damages with my bad coding for today. 

Bugfix on addchild. Parent is now assigned properly when multiple chidren are added at once.

Bugfix on reload. Reloading transient properties was causing their value to be overwritten by `null` or empty. Now they are skipped during reload.

Bugfix on checkAnimalsForClones. It wasn't warking at all. I ended up rewriting it simplifying it a bit.

Added lookForData function. While loading if no children are added to the loaded obj (because none ere found) the user is prompted to select the correct data folder. This triggers a path update for the nigelObj (and subsequently loaded children).

Nabarb committed:
nigelObj improvements

Bugfix on addchild. Parent is now assigned properly when multiple chidren are added at once.

Bugfix on reload. Reloading transient properties was causing their value to be overwritten by `null` or empty. Now they are skipped during reload.

Bugfix on checkAnimalsForClones. It wasn't warking at all. I ended up rewriting it simplifying it a bit.

Added lookForData function. While loading if no children are added to the loaded obj (because none ere found) the user is prompted to select the correct data folder. This triggers a path update for the nigelObj (and subsequently loaded children).

m053m716 commented:
Will try and look this over tonight. Depending on how it looks I might need one more day to go over it, but hopefully try and get through it quickly so you can keep doing Pull Requests- thanks!

## To Do ##
- [ ] Triple-check `checkAnimalsForClones` 
- [ ] Double-check `lookForData` (prompts user to select correct Block location now)
- [ ] Double-check `reload` (no longer assigns null property improperly)
- [ ] Double-check `addChild` (adds Parents properly now)

m053m716 committed:
Update documentation

Added brief documentation to `lookForData` method.

m053m716 commented on +nigeLab/@nigelObj/nigelObj.m:
```diff
@@ -5055,45 +5067,15 @@ function checkAnimalsForClones(obj,~,~)
-            addChild(a,B);
-            
-            % Now, remove redundant animals from array and also remove them
-            % from the comparisons matrix since we don't need to redo them
-            mask = find(idx);
-            obj.Children(mask(animalIsSame)) = []; % Remove from property
-            A(animalIsSame) = []; % Remove them from consideration in the array
-            idx(animalIsSame) = []; % Remove corresponding indexes
-            
-            % Lastly, update the comparisons matrices
-            iRow = animalIsSame(2:end); % To account for previously-removed row of comparisons
-            comparisons_mat(iRow,:) = [];
-            % Columns are not removed, since the original animal is kept in
-            % the array and we should account for its index.
-            comparisons_mat(:,animalIsSame) = [];
+         while any(rmvec)             
+             idxToRemove = find(rmvec,1);
+             idxToKeep = find(correspondanceVec,1);
+             a = A(idxToKeep);
+             B = [A(idxToRemove).Children]; %#ok<*FNDSB>
+             addChild(a,B);
+             obj.Children(idxToRemove) = []; % Remove from property
+             rmvec(idxToRemove) = [];
+             correspondanceVec(idxToKeep) = [];
```
I remember having issues with this when I initially wrote it, then re-wrote it to try and simplify it, then eventually reverted back to the older "less-elegant" code. Have you tested this out on different use-cases? One issue could have been during `reload` if we were reloading transient Animal objects as `null`, then this would automatically remove them.

Nabarb commented:
There is an iddue with lookForData. At this point it is called when no children are found. If a tank is moved from a pc to another though it might try to save a .nigelTank at the old location. This might end up with an `Access Denied` error

Nabarb committed:
Fix AccessDenied error

Fixes #82

Nabarb committed:
Dashboard opening issues fixed

Commented out some leftover chuncks regarding multiAnimals staff that were causing problems here

Nabarb committed:
Fixed Add dashboard button

Small fixes to get the Add button to work
